internal/provider/resource_hvn_route.go

```diff
@@ -1,14 +1,275 @@
 package provider
 
 import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
 	"time"
 
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/client/network_service"
 	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
 	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
 )
 
 var hvnRouteDefaultTimeout = time.Minute * 1
+var hvnRouteCreateTimeout = time.Minute * 35
+
+func resourceHvnRoute() *schema.Resource {
+	return &schema.Resource{
+		Description: "The HVN route resource allows you to manage an HVN route.",
+
+		CreateContext: resourceHvnRouteCreate,
+		ReadContext:   resourceHvnRouteRead,
+		DeleteContext: resourceHvnRouteDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Default: &hvnRouteDefaultTimeout,
+			Create:  &hvnRouteCreateTimeout,
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceHVNRouteImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			// Required inputs
+			"hvn": {
+				Description: "The `self_link` of the HashiCorp Virtual Network (HVN).",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"hvn_route_id": {
+				Description:      "The ID of the HVN route.",
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validateSlugID,
+			},
+			"destination_cidr": {
+				Description:  "The destination CIDR of the HVN route",
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IsCIDR,
+			},
+			"target_link": {
+				Description: "A unique URL identifying the target of the HVN route.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			// Computed outputs
+			"self_link": {
+				Description: "A unique URL identifying the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"state": {
+				Description: "The state of the HVN route.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"created_at": {
+				Description: "The time that the HVN route was created.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			// TODO add updated_at after public API returns updated_at for HVN route
```

Comment:
Please see https://github.com/hashicorp/terraform-provider-hcp/pull/122/files#r633792131

Reply:
this PR didn't add `updated_at` for hvn routes, so we don't expect it to be added, removed that TODO